### DISCUSSION TITLE
Help with regex in Importer

### DISCUSSION THREAD

**Discussion by e-zaline** [Spine-project/Spine-Toolbox#2889] [Support]
Hello, sorry for this basic question, but I want to filter with regex in my SpineToolbox importer.
I want to keep only strings that have **two** letters + "NG" (e.g. XX**NG**XX, LY**NG**CC_Misurata_CC, etc...), and filter out the other strings, for instance strings like XXXNG (i.e. **more than two letters** before NG).

I thought this was the way (it's what I understood from [wikipedia](https://en.wikipedia.org/wiki/Regular_expression))
"[A-Za-z][A-Za-z]NG"
<img width="427" alt="image" src="https://github.com/user-attachments/assets/0b266a9a-e5a9-4857-b2f3-6d3aa0279510">

But this does not filter out the XXXXNG.

What would be the correct way in SpineToolbox?

**Comment by soininen** [comment]
Regular expression are very powerful but oftentimes complicated to get right. "[A-Za-z][A-Za-z]NG" matches strings like XXXXNG because it still contains the pattern XXNG in the middle. What you need is a little token `^` that matches the beginning of a string: `^[A-Za-z][A-Za-z]NG`

There are handy pages like [this](https://regex101.com/) that you can use to build and test your regular expressions.

**Reply by e-zaline** [reply]
Thank you for the quick reply Antti! 👍
